cmd/cli/config/util.go

```diff
@@ -0,0 +1,59 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+)
+
+// parseStringSliceToMap parses a slice of strings into a map.
+// Each element in the slice should be a key-value pair in the form "key=value".
+func parseStringSliceToMap(slice []string) (map[string]string, error) {
+	result := make(map[string]string)
+	for _, item := range slice {
+		parts := strings.SplitN(item, "=", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("expected 'key=value', receieved invalid format for key-value pair: %s", item)
+		}
+		key, value := parts[0], parts[1]
+		result[key] = value
+	}
+	return result, nil
+}
+
+func singleValueOrError(v ...string) (string, error) {
+	if len(v) != 1 {
+		return "", fmt.Errorf("expected single value got %d from %q", len(v), v)
+	}
+	return v[0], nil
+}
+
+// TODO for the node type methods below. Make the NodeType a concrete type so we can make type assertions on it.
```

Comment:
We already have `Requester` and `Compute` types in pkg/node which are disjoint types (they should probably be implementations of some interface). I'm not sure we need to add more abstractions, we could just use `getNodeType` which is in cmd/cli/serve/util.go after changing it to take a slice of strings (rather than allow it to dig into config itself).

Reply:
Removed the TODO.